### PR TITLE
feat: add version control for tooling

### LINKED ISSUE #1099: chore: add version control to all the rust tooling
Currently we use `rustc`, `cargo`, `clippy` `rustfmt` and `prettier` as our tools. The setup that we have right now doesn't support maintaining a version in the repository and thus your local lint tools version could be different than what's running on CI.

**Proposal**
- Have the ability to maintain consistent versions between the CI and local machine for all the above tools.

### PR DESCRIPTION
**Pull request by ologbonowiwi**
- chore: lock cargo/rust version
- chore: lock rust fmt version
- chore: lock prettier and clippy versions

**Issue Reference(s):**  
Fixes #1099
/claim #1099
/split @neo773 50%

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [x] I have performed a self-review of my code.
- [x] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ jobs:
         with:
           node-version: "20.11.0"
       - name: Install Prettier
-        run: npm install --global prettier
+        run: npm install --global prettier@3.2.5
```
Comment (ologbonowiwi) [inline .github/workflows/lint.yml:33]:
```suggestion
```

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -30,7 +30,7 @@ jobs:
         with:
           node-version: "20.11.0"
       - name: Install Prettier
```
Comment (ologbonowiwi) [inline .github/workflows/lint.yml:32]:
```suggestion
```

File: .github/workflows/lint.yml (hunk 1 above)
Comment (ologbonowiwi) [inline .github/workflows/lint.yml:33]:
```suggestion
```

File: lint.sh [hunk 3]
```diff
@@ -27,9 +27,9 @@ run_cargo_clippy() {
 run_prettier() {
     MODE=$1
     if [ "$MODE" == "check" ]; then
-        prettier -c .prettierrc --check "**/*.$FILE_TYPES"
+        npx --yes prettier@3.2.5 -c .prettierrc --check "**/*.$FILE_TYPES"
```
Comment (amitksingh1490) [inline lint.sh:30]: adding version to scripts will not allow renovate to open update PRs

Comment (ologbonowiwi) [reply lint.sh:30]: we can add a package.json on the project root to lock the npm dependency, in this way renovate will be able to update it but still users will be able to use the expected version

Comment (webbdays) [reply lint.sh:30]: any progress on this?

Comment (ologbonowiwi) [reply lint.sh:30]: The last update was [this suggestion](https://github.com/tailcallhq/tailcall/issues/1099#issuecomment-1937145466), but @neo773 tried, and I think he had some issues with node dependencies on Nix.

**Pull request closed without merging**